src/System.Device.Gpio/System/Device/Pwm/Channels/UnixPwmChannel.Linux.cs

```diff
@@ -0,0 +1,181 @@
+        private int GetPeriodInNanoSeconds(double frequency) => (int)((1.0 / frequency) * 1_000_000_000);
+
+        /// <summary>
+        /// Sets the polarity for the channel to start in.
+        /// </summary>
+        /// <param name="polarity">The polarity to set.</param>
+        private void SetPolarity(PwmPolarity polarity)
+        {
+            string polarityString = polarity switch
+            {
+                PwmPolarity.ActiveHigh => "normal",
+                PwmPolarity.ActiveLow => "inversed",
+                _ => throw new ArgumentException($"Invalid PWM polarity {polarity}.")
+            };
+
+            File.WriteAllText($"{GetChannelPath()}/polarity", Convert.ToString(polarityString));
+        }
+
+        /// <summary>
+        /// Sets the frequency for the channel.
+        /// </summary>
+        /// <param name="frequency">The frequency in hertz to set.</param>
+        private void SetFrequency(double frequency)
+        {
+            int periodInNanoSeconds = GetPeriodInNanoSeconds(frequency);
+            File.WriteAllText($"{GetChannelPath()}/period", Convert.ToString(periodInNanoSeconds));
+            _frequency = frequency;
+        }
+
+        /// <summary>
+        /// Sets the duty cycle percentage for the channel.
+        /// </summary>
+        /// <param name="dutyCyclePercentage">The duty cycle percentage to set represented as a value between 0.0 and 1.0.</param>
+        private void SetDutyCyclePercentage(double dutyCyclePercentage)
+        {
+            if (dutyCyclePercentage < 0 || dutyCyclePercentage > 1)
+            {
+                throw new ArgumentOutOfRangeException(nameof(dutyCyclePercentage), dutyCyclePercentage, "Value must be between 0.0 and 1.0.");
+            }
+
+            // In Linux, the period needs to be a whole number and can't have decimal point.
+            int dutyCycleInNanoSeconds = (int)(GetPeriodInNanoSeconds(_frequency) * dutyCyclePercentage);
+            File.WriteAllText($"{GetChannelPath()}/duty_cycle", Convert.ToString(dutyCycleInNanoSeconds));
+            _dutyCyclePercentage = dutyCyclePercentage;
+        }
+
+        /// <summary>
+        /// Closes the channel.
+        /// </summary>
+        private void CloseChannel()
+        {
+            if (Directory.Exists(GetChannelPath()))
+            {
+                File.WriteAllText($"{GetChipPath()}/unexport", Convert.ToString(_pwmChannel));
+            }
+        }
+
+        /// <summary>
+        /// Opens the channel.
+        /// </summary>
+        private void OpenChannel()
+        {
+            if (!Directory.Exists(GetChannelPath()))
+            {
+                File.WriteAllText($"{GetChipPath()}/export", Convert.ToString(_pwmChannel));
+            }
+        }
+
+        /// <summary>
+        /// Starts writing to the channel.
+        /// </summary>
+        public override void Start()
```

Comment:
The original plan was to have StartWriting/StopWriting, but it seems like Start/Stop sounds better.  It is also the same method names in the WinRT PWM API.  Linux uses Enable/Disable.  I like any of them, so let me know the names and will update.

@krwq You mentioned not having Start/Stop in the end.  We can still remove, but I thought of a scenario where a user might want to stop the PWM (without disposing) while making DC adjustments (especially if there were multiple channels) and then start back.  The channel is opened when it is instantiated and closed when disposed.